Dockerfile

```diff
@@ -1,11 +1,14 @@
 # Install the app dependencies in a full Node docker image
 FROM registry.access.redhat.com/ubi8/nodejs-16:latest
 
-# Copy package.json and package-lock.json
-COPY package*.json ./
+# Copy package.json, and optionally package-lock.json if it exists
+COPY package.json package-lock.json* ./
 
 # Install app dependencies
-RUN npm ci
+RUN \
+  if [ -f package-lock.json ]; then npm ci; \
+  else npm install; \
+  fi
```

Comment:
Fallback in case of no `package-lock.json`. It's unusual for a Node.js project repository to not have one these days, but let's cater for the exception.